pkg/providers/snow/fetch.go

```diff
@@ -0,0 +1,38 @@
+package snow
+
+import (
+	"context"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	snowv1 "github.com/aws/eks-anywhere/pkg/providers/snow/api/v1beta1"
+)
+
+func oldControlPlaneMachineTemplate(ctx context.Context, kubeClient KubeClient, oldSpec *cluster.Spec) (*snowv1.AWSSnowMachineTemplate, error) {
+	cp := &controlplanev1.KubeadmControlPlane{}
+	if err := kubeClient.Get(ctx, oldSpec.Cluster.GetName(), constants.EksaSystemNamespace, cp); err != nil {
```

Comment:
could we have a single method that we use to derive the the `KubeadmControlPlane` from a `clusterSpec`? That way we don't need to know here that kubeadmCP are named after the eksa cluster

Or maybe we should do it with labels?